### DISCUSSION TITLE
OMP doesn't seem to load by default in the built in terminal in VSCode

### DISCUSSION THREAD

**Discussion by xela-nosretap** [JanDeDobbeleer/oh-my-posh#1397] [Troubleshoot] [answered]
I have OMP installed in Windows and when I load up Windows Terminal and the terminal in VSCode OMP loads fine.

I also have Debian installed via WSL. If I open Windows Terminal and open the Debian tab OMP loads fine, if I open Debian from the start menu a terminal window opens and OMP loads fine, if I'm in VSCode pointed to a WSL folder and open the terminal OMP doesn't load. I need to type in `. ~/.profile` to get OMP to load.

I don't think this is an OMP issue, more an issue with how I've set something up (probably in VSCode) but I'm not sure what it could be.

Does anyone have any thoughts?

**Comment by JanDeDobbeleer** [comment]
I would ask that question to the VSCode team, my assumption is they load a different profile (or only the `.bashrc` file maybe?)

**Comment by xela-nosretap** [comment, marked as answer]
Just in case anyone has this issue in future I managed to fix it by adding:
```
"terminal.integrated.profiles.linux": {
        "bash": {
          "path": "bash",
          "args": ["-l"]
        }
      }
```
into my `settings.json` file.

**Comment by akelge** [comment]
Well, I think you can move the command to launch OMP into .bashrc, that should be processed in any case, while .profile is processed only on login shells, IIRC, hence the need for `-l`
